moved/src/move_execution/canonical.rs

```diff
@@ -153,13 +166,21 @@ pub(super) fn execute_canonical_transaction(
         }
     };
 
+    let gas_used = total_gas_used(&gas_meter, genesis_config);
+    base_token.refund_gas_cost(
```

Comment:
Refunding cause test failures. I think there is a logical issue here. We compute the `gas_used` but we haven't charged it yet, so refunding it before charging seems wrong.

Reply:
It's actually due to me setting it to zero, and it was properly charged up to a limit [here](https://github.com/MovedNetwork/op-move//blob/feat/ilchu/l2_gas_calc/moved/src/move_execution/canonical.rs#L74-L80). The issue with the tests was due to L2 costs being set by me to 0 in tests, whereas actual gas meter did accrue some units, and it was failing trying to refund back the sum that wasn't properly deposited into treasury in the first place.